cli/src/commands/tag.rs

```diff
@@ -69,3 +88,43 @@ fn cmd_tag_list(
 
     Ok(())
 }
+
+fn cmd_tag_forget(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &TagForgetArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let mut tx = workspace_command.start_transaction();
+    let view = tx.repo().base_repo().view();
+
+    let mut to_forget = vec![];
+    let mut unmatched_patterns = vec![];
+    for pattern in args.names.iter() {
+        let mut matched = false;
+        for tagname in view.tags().keys().filter(|name| pattern.matches(name)) {
+            matched = true;
+            to_forget.push(tagname.clone());
+        }
+        if !matched {
+            unmatched_patterns.push(pattern.to_string())
+        }
+    }
+    if !unmatched_patterns.is_empty() {
+        return Err(user_error(format!(
+            "No matching tags for patterns: {}",
+            unmatched_patterns.into_iter().join(", ")
+        )));
+    }
+    for tagname in to_forget.iter() {
+        // Forgetting a tag is the same as deleting it at the moment, as we never push
+        // tags.
+        tx.mut_repo().set_tag_target(tagname, RefTarget::absent());
+    }
+    tx.finish(ui, format!("forgot {} tags", to_forget.len()))?;
```

Comment:
nit: `forget` (present tense)

and maybe better to include names `to_forget.iter().join(", ")`.